app/assets/javascripts/discourse/components/d-editor.js.es6

```diff
@@ -636,23 +638,44 @@ export default Ember.Component.extend({
     return null;
   },
 
+  @computed
+  pastePlaceholder() {
+    return I18n.t('pasting');
+  },
+
   paste(e) {
     const clipboard = clipboardData(e);
     const types = clipboard.types;
-    let preventDefault = false;
+    const plainText = clipboard.getData("text/plain");
+    const placeholder = plainText || this.get('pastePlaceholder');
+    let handled = false;
 
-    if (types.some(t => t === "text/plain")) {
-      const text = clipboard.getData("text/plain");
-      const table = this._detectTable(text);
+    if (plainText) {
+      const table = this._detectTable(plainText);
       if (table) {
         this._addText(this._getSelected(), table);
-        preventDefault = true;
+        handled = true;
       }
-    } else if (types.some(t => t === "Files")) {
-      preventDefault = true;
     }
 
-    if (preventDefault) {
+    if (!handled && types.some(t => t === "text/html")) {
+      const html = clipboard.getData("text/html");
+      const self = this;
+      ajax('/composer/parse_html', {
+        type: 'POST',
+        data: { html }
+      }).then(response => {
+        self.appEvents.trigger('composer:replace-text', placeholder, response.markdown);
+      }).catch(error => {
+        if (!plainText) {
+          popupAjaxError(error);
```

Comment:
If something fails, why not simply paste in the text in non html, not sure if reporting the error to the user actually helps

Reply:
I am reporting only if the `plainText` is not available. If it is available it will be already pasted.